File: apps/api/src/Procraft.Api/Controllers/HealthController.cs
using Microsoft.AspNetCore.Mvc;

namespace Procraft.Api.Controllers;

[ApiController]
[Route("api/health")]
public sealed class HealthController : ControllerBase
{
    /// <summary>Operational health reachable through the /api prefix for gateway routing symmetry.</summary>
    [HttpGet]
    public IActionResult Get()
    {
        return Ok(new { status = "Healthy" });
    }
}

File: apps/api/src/health/health.controller.ts
import { Controller, Get } from '@nestjs/common';

/** /health (gateway probe) and /api/health (proxied symmetry), like the C# host. */
@Controller()
export class HealthController {
  @Get('health')
  health() {
    return { status: 'Healthy' };
  }

  @Get('api/health')
  apiHealth() {
    return { status: 'Healthy' };
  }
}
